File: restaurantservice/employee-service/src/main/java/pl/restaurant/employeeservice/api/response/EmployeeInfo.java
package pl.restaurant.employeeservice.api.response;

import lombok.*;
import pl.restaurant.employeeservice.api.request.Address;

import java.math.BigDecimal;
import java.time.LocalDate;
import java.util.List;

@Getter
@Setter
@Builder
@AllArgsConstructor
@NoArgsConstructor
public class EmployeeInfo {
    private EmployeeShortInfo shortInfo;
    private String phoneNr;
    private String pesel;
    private String accountNr;
    private BigDecimal salary;
    private boolean active;
    private LocalDate employmentDate;
    private LocalDate dismissalDate;
    private Address address;
    private RestaurantShortInfo restaurantInfo;
    private List<ScheduleInfo> schedulesInfo;
}

File: restaurant-app/src/app/model/employee/employee-info.ts
import {EmployeeShortInfo} from "./employee-short-info";
import {Address} from "../address";
import {RestaurantShortInfo} from "../restaurant/restaurant-short-info";
import {ScheduleInfo} from "../schedule/schedule-info";

export interface EmployeeInfo {
  shortInfo: EmployeeShortInfo;
  phoneNr: string;
  pesel: string;
  accountNr: string;
  salary: number;
  active: boolean;
  employmentDate: Date | any;
  dismissalDate: any;
  address: Address;
  restaurantInfo: RestaurantShortInfo;
  schedulesInfo: ScheduleInfo[];
}
